File: epi_judge_java/epi/TreePostorder.java
package epi;
import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;
public class TreePostorder {

  private static class NodeAndState {
    public BinaryTreeNode<Integer> node;
    public Boolean subtreesTraversed;

    public NodeAndState(BinaryTreeNode<Integer> node,
                        Boolean subtreesTraversed) {
      this.node = node;
      this.subtreesTraversed = subtreesTraversed;
    }
  }

  @EpiTest(testDataFile = "tree_postorder.tsv")
  public static List<Integer> postorderTraversal(BinaryTreeNode<Integer> tree) {
    // TODO - you fill in here.
    return null;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "TreePostorder.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi_judge_python/tree_postorder.py
from typing import List

from binary_tree_node import BinaryTreeNode
from test_framework import generic_test


def postorder_traversal(tree: BinaryTreeNode) -> List[int]:
    # TODO - you fill in here.
    return []


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('tree_postorder.py',
                                       'tree_postorder.tsv',
                                       postorder_traversal))
